lib/output.js

```diff
@@ -1283,13 +1283,17 @@ function OutputStream(options) {
     };
 
     // self should be AST_New.  decide if we want to show parens or not.
-    function no_constructor_parens(self, output) {
-        return self.args.length == 0 && !output.option("beautify") ||
-            !(self.expression instanceof AST_SymbolRef ||
+    function need_constructor_parens(self, output) {
+        // Always print parentheses with arguments
+        if (self.args.length > 0) return true;
+
+        // Certain expressions are safe to add parentheses
+        return output.option("beautify") && (
+              self.expression instanceof AST_SymbolRef ||
               self.expression instanceof AST_Call ||
               self.expression instanceof AST_Function ||
```

Comment:
Well it can, as all members will behave the same. So then this patch should be able to fail/succeed on `AST_Accessor`, `AST_Function` and `AST_Defun`.